### ISSUE TITLE
Inconsistent metadata after multiplication

### ISSUE THREAD

**Issue by kayibal** [dask/dask#6441]
**What happened**:
If a dask.DataFrame is multiplied with a pandas.Series, the metadata becomes inconsistent with the actual data.

**What you expected to happen**:
Metadata should be updated accordingly.

**Minimal Complete Verifiable Example**:

```python
import dask.dataframe as dd
import pandas as pd
import numpy as np

df = pd.DataFrame(np.ones((10, 3)), columns=list('BCA'))
factors = pd.Series([1, 2, 3], index=list('ABC'))

ddf = dd.from_pandas(df, 1)
ddf_mul = ddf * factors
res = ddf_mul.map_partitions(lambda x: x)
res.compute()
```
This will raise the following error:
```
ValueError: The columns in the computed data do not match the columns in the provided metadata
Order of columns does not match
```

**Environment**:

- Dask version: 2.21.0
- Python version: 3.7
- Operating System: macOS
- Install method (conda, pip, source): pip

**Comment by quasiben** [comment]
To resolve issues like these you can supply a `meta` argument to map_partitions to help dask understand what the results output data should look like. In your case something like the following should work:

```python
In [32]: meta = pd.DataFrame({'A': [], 'B': [], 'C': []})

In [33]: ddf_mul.map_partitions(lambda x: x, meta=meta).head()
```

**Comment by jsignell** [comment]
Another approach would be to make sure that `_meta` is correct on `ddf_mul` itself. One way to do that is to replace:

```python
ddf_mul = ddf * factors
```

with 
```python
ddf_mul = ddf.mul(factors)
```
I have an idea for how we could make this work better with the example you provided though. Pull request coming soon!

**Comment by quasiben** [comment]
closed by #6447

**Comment by jsignell** [comment]
Note that #6447 was closed not merged. But I think it's ok to close this issue anyways since passing the meta works.
